src/Monolog/Formatter/LineFormatter.php

```diff
@@ -204,16 +204,19 @@ protected function normalizeException(\Throwable $e, int $depth = 0): string
     {
         $str = $this->formatException($e);
 
-        if (($previous = $e->getPrevious()) instanceof \Throwable) {
-            do {
-                $depth++;
-                if ($depth > $this->maxNormalizeDepth) {
-                    $str .= "\n[previous exception] Over " . $this->maxNormalizeDepth . ' levels deep, aborting normalization';
-                    break;
-                }
+            while (true) {
+            $previous = $e->getPrevious();
```

Comment:
this is broken, because you restart with `$previous = $e->getPrevious();` at each iteration of the loop instead of using the modified `$previous`